daemon/routes/iprule/iprule_test.go

```diff
@@ -183,3 +186,62 @@ func TestCustomTable(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Greater(t, prioID, prioID2)
 }
+
+func TestAddAllowlistRules(t *testing.T) {
+	category.Set(t, category.Route)
+
+	rulesBeforeAdd, err := netlink.RuleList(netlink.FAMILY_V4)
+	assert.NoError(t, err, "Failed to save previous state for ip rules: %s", err)
+
+	router := Router{
+		subnetToRulePriority: make(map[string]uint),
+	}
+
+	allowlist := []string{
+		// two /0 subnets should be simplified to a single default route
+		"35.74.174.235/0",
+		"4.246.215.86/0",
+		"103.238.215.35/24",
+		"237.164.3.235/32",
+	}
+
+	err = router.addAllowlistRules(allowlist, false)
+	assert.NoError(t, err, "Unexpected error when adding allowlist rules: %s", err)
+
+	rulesAfterAdd, err := netlink.RuleList(netlink.FAMILY_V4)
+
+	assert.NoError(t, err, "Failed to save new state for ip rules: %s", err)
+	// ignore rules that existed before the test for validation purposes
+	rulesAfterAdd = slices.DeleteFunc(rulesAfterAdd, func(rule netlink.Rule) bool {
+		idx := slices.Index(rulesBeforeAdd, rule)
+		return idx != -1
+	})
+
+	_, subnet2, _ := net.ParseCIDR("103.238.215.35/24")
```

Comment:
https://pkg.go.dev/github.com/projectcalico/libcalico-go/lib/net#MustParseCIDR can be used in tests. Or, you can even construct the struct directly without a need to parse it